sbin/pack.sh

```diff
@@ -0,0 +1,224 @@
+		DEPS=0|1          Build dependencies file
+		SYM=0|1           Build debug symbols file
+
+		VARIANT=name      Build variant (default: empty)
+		BRANCH=name       Branch name for snapshot packages
+		GITSHA=1          Append Git SHA to shapshot package names
+
+		ARTDIR=dir        Directory in which packages are created (default: bin/artifacts)
+		BINDIR=dir        Directory in which binaries are found
+
+		VERBOSE=1         Print commands
+		IGNERR=1          Do not abort on error
+
+	END
+	exit 0
+fi
+
+#----------------------------------------------------------------------------------------------
+
+[[ $IGNERR == 1 ]] || set -e
+[[ $V == 1 || $VERBOSE == 1 ]] && set -x
+
+RAMP=${RAMP:-1}
+DEPS=${DEPS:-0}
+SYM=${SYM:-0}
+
+[[ -z $BINDIR ]] && BINDIR=$ROOT/bin/$($READIES/bin/platform --os)-$($READIES/bin/platform --arch)-release
+BINDIR=$(cd $BINDIR && pwd)
+
+[[ -z $ARTDIR ]] && ARTDIR=bin/artifacts
+mkdir -p $ARTDIR $ARTDIR/snapshots
+ARTDIR=$(cd $ARTDIR && pwd)
+
+. $READIES/bin/enable-utf8
+
+export ARCH=$($READIES/bin/platform --arch)
+export OS=$($READIES/bin/platform --os)
+export OSNICK=$($READIES/bin/platform --osnick)
+
+# RLEC naming conventions
+[[ $ARCH == x64 ]] && ARCH=x86_64
+[[ $OS == linux ]] && OS=Linux
+if [[ $OSNICK == focal ]]; then
+	OSNICK=ubuntu20.04
+elif [[ $OSNICK == bionic ]]; then
+	OSNICK=ubuntu18.04
+elif [[ $OSNICK == xenial ]]; then
+	OSNICK=ubuntu16.04
+elif [[ $OSNICK == trusty ]]; then
+	OSNICK=ubuntu14.04
+elif [[ $OSNICK == centos7 ]]; then
+	OSNICK=rhel7
+elif [[ $OSNICK == centos8 ]]; then
+	OSNICK=rhel8
+fi
+
+export PRODUCT=redistimeseries
+export PRODUCT_LIB=$PRODUCT.so
+export DEPNAMES=""
+
+export PACKAGE_NAME=${PACKAGE_NAME:-${PRODUCT}}
+
+RAMP_CMD="python3 -m RAMP.ramp"
+
+#----------------------------------------------------------------------------------------------
+
+pack_ramp() {
+	cd $ROOT
+
+	local platform="$OS-$OSNICK-$ARCH"
+	local stem=${PACKAGE_NAME}.${platform}
+
+	local verspec=${SEMVER}${VARIANT}
+	
+	local fq_package=$stem.${verspec}.zip
+
+	local packfile="$ARTDIR/$fq_package"
+	local product_so="$BINDIR/$PRODUCT.so"
+
+	local xtx_vars=""
+	local dep_fname=${PACKAGE_NAME}.${platform}.${verspec}.tgz
+
+	if [[ -z $VARIANT ]]; then
+		local rampfile=ramp.yml
+	else
+		local rampfile=ramp$VARIANT.yml
+	fi
+
+	python3 $READIES/bin/xtx \
+		$xtx_vars \
+		-e NUMVER -e SEMVER \
+		$ROOT/$rampfile > /tmp/ramp.yml
+	rm -f /tmp/ramp.fname $packfile
+	$RAMP_CMD pack -m /tmp/ramp.yml --packname-file /tmp/ramp.fname --verbose --debug -o $packfile $product_so >/tmp/ramp.err 2>&1 || true
+	if [[ ! -e $packfile ]]; then
+		eprint "Error generating RAMP file:"
+		>&2 cat /tmp/ramp.err
+		exit 1
+	fi
+
+	cd $ARTDIR/snapshots
+	if [[ ! -z $BRANCH ]]; then
+		local snap_package=$stem.${BRANCH}${VARIANT}.zip
+		ln -sf ../$fq_package $snap_package
+	fi
+
+	cd $ROOT
+}
+
+#----------------------------------------------------------------------------------------------
+
+pack_deps() {
```

Comment:
@rafie is this a new generic pack script? we don't really have deps (all our deps are compiled in).

Reply:
@danni-m Yes, we're converging in all modules to a common packing logic.